vmdb/app/controllers/mixins/containers_common_mixin.rb

```diff
@@ -47,6 +47,50 @@ def show_container(record, controller_name, display_name)
+                      :url  => "/#{controller_name}/show/#{record.id}?display=#{@display}")
+      @view, @pages = get_view(ContainerService, :parent => record)  # Get the records (into a view) and the paginator
+      @showtype = @display
+      if @view.extras.fetch(:total_count, 0) > @view.extras.fetch(:auth_count, 0)
+        @bottom_msg = "* You are not authorized to view "
+        + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}")
+        + " on this " + ui_lookup(:tables => @table_name)
+      end
+    elsif @display == "container_routes" || session[:display] == "container_routes" && params[:display].nil?
+      title = ui_lookup(:tables => "container_routes")
+      drop_breadcrumb(:name => record.name + " (All #{title})",
+                      :url  => "/#{controller_name}/show/#{record.id}?display=#{@display}")
+      @view, @pages = get_view(ContainerRoute, :parent => record)  # Get the records (into a view) and the paginator
+      @showtype = @display
+      if @view.extras.fetch(:total_count, 0) > @view.extras.fetch(:auth_count, 0)
+        @bottom_msg = "* You are not authorized to view "
+        + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}")
+        + " on this " + ui_lookup(:tables => @table_name)
+      end
+    elsif @display == "container_replicators" || session[:display] == "container_replicators" && params[:display].nil?
+      title = ui_lookup(:tables => "container_replicators")
+      drop_breadcrumb(:name => record.name + " (All #{title})",
+                      :url  => "/#{controller_name}/show/#{record.id}?display=#{@display}")
+      @view, @pages = get_view(ContainerReplicator, :parent => record) # Get the records (into a view) and the paginator
+      @showtype = @display
+      if @view.extras.fetch(:total_count, 0) > @view.extras.fetch(:auth_count, 0)
+        @bottom_msg = "* You are not authorized to view "
+        + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}")
+        + " on this " + ui_lookup(:tables => @table_name)
+      end
+    elsif @display == "container_projects" || session[:display] == "container_projects" && params[:display].nil?
+      title = ui_lookup(:tables => "container_projects")
+      drop_breadcrumb(:name => record.name + " (All #{title})",
+                      :url  => "/#{controller_name}/show/#{record.id}?display=#{@display}")
+      @view, @pages = get_view(ContainerProject, :parent => record)  # Get the records (into a view) and the paginator
+      @showtype = @display
+      if @view.extras.fetch(:total_count, 0) > @view.extras.fetch(:auth_count, 0)
+        @bottom_msg = "* You are not authorized to view "
+        + pluralize(@view.extras[:total_count] - @view.extras[:auth_count], "other #{title.singularize}")
+        + " on this " + ui_lookup(:tables => @table_name)
+      end
```

Comment:
This looks like a lot of similar code.  Can it be refactored?  Are there any new tests/specs for this chunk of code?

cc @dclarizio @h-kataria @martinpovolny